fedbiomed/common/data/_sklearn_data_manager.py

```diff
@@ -349,7 +352,12 @@ def split(self, test_ratio: float) -> Tuple[NPDataLoader, NPDataLoader]:
             self._subset_test = (x_test, y_test)
             self._subset_train = (x_train, y_train)
 
-        test_batch_size = max(1, len(self._subset_test[0]))
+        if not test_batch_size:
+            # case where test_batch_size=0 or test_batch_size=None, ie where user doesnot want
+            # to use test_batch_size
+            test_batch_size = len(self._subset_test)
```

Comment:
That could be a nice idea, but I thought it would have gone beyond the scope of this PR. Basically your idea is to set a threshold to something like `limit`
```python
limit = 1
for elem in dataset.shape:
     limit *= elem
```
But usually dataset shape is `[dataset size, 1st dim, 2nd dim, ...]`
for instance for images: `[1000, 28, 28]`, limit would be `limit = 784`, is that what you were thinking` ? Take the MNIST dataset,  you could use the whole dataset instead of setting a limit. 

In my opinion `test_batch_size` should be used in very specific and rare cases, therefore i don't think it could be that useful to have a limit.

Reply:
Hi @ybouilla, 

I agree that batch size can be used in very specific cases. What I mean is that using test batch size is decided by researcher and it is not safe since it will block the memory on machine that node is running if researcher does not set it. This may create other bugs. For example, training fails due to memory issue but at the same time there may be another request coming to node (eg dataset search) it does not work because the memory is full. 

Therefore, node can set a limit for batch size to avoid such issues. This issues is also existing for training, and I agree that it is beyond the scope of this PR. It could be nice to create an issue to investigate it more.

Reply:
Ok what i wanted to point out is that even if your idea is nice,  it is not something easy to do. Understood, let's open an issue for that!